pkg/fieldpath/paved.go

```diff
@@ -404,6 +405,12 @@ func (p *Paved) setValue(s Segments, value any) error {
 				return nil
 			}
 
+			// If the next segment is an index we need to ensure the field is not above the max index, if set.
```

Comment:
Isn't this case already covered with the above check 🤔 I mean case `SegmentIndex` ?